include/aws/http/connection.h

```diff
@@ -35,6 +35,94 @@ typedef void(
 typedef void(
     aws_http_on_client_connection_shutdown_fn)(struct aws_http_connection *connection, int error_code, void *user_data);
 
+/*
+ * Connection target properties
+ */
+struct aws_http_endpoint_options {
+
+    /*
+     * Required
+     */
+    struct aws_byte_cursor schema;
+
+    /**
+     * Required.
+     */
+    struct aws_byte_cursor host_name;
```

Comment:
It might be simpler to just use `aws_uri`?
Rather than declare a unique struct.
I'm super conflicted on this